src/main/java/ui/UI.java

```diff
@@ -245,11 +244,17 @@ private void initPreApplicationState() {
     private void initApplicationState() {
         // In the future, when more arguments are passed to logic,
         // we can pass them in the form of an array.
+
         logic = new Logic(uiManager, prefs, Optional.empty(), Optional.empty());
+
         // TODO clear cache if necessary
-        refreshTimer = new TickingTimer("Refresh Timer", REFRESH_PERIOD,
-                                        status::updateTimeToRefresh, logic::refresh, TimeUnit.SECONDS);
-        refreshTimer.start();
+        
+        TickingTimer refreshTimer = new TickingTimer("Refresh Timer", TickingTimerManager.DEFAULT_REFRESH_PERIOD_IN_SEC,
+            status::updateTimeToRefresh, logic::refresh, TimeUnit.SECONDS);
+
+        timerManager = new TickingTimerManager(refreshTimer);
+        timerManager.startTimer();
```

Comment:
Maybe this could be named `RefreshTimer`, and could encapsulate/subclass the timer object? In an OO sense it's a more specific kind of timer which knows about refresh intervals and such. Following the same interface is a plus.

The responsibility of a 'Manager' is a bit too vague here IMO.